client/app/dbaas/logs/detail/streams/add/logs-streams-add.html

```diff
@@ -1,4 +1,148 @@
+                                id="stream_archive_algorithm"
+                                data-ng-model="ctrl.stream.data.coldStorageCompression"
+                                data-ng-options="algo.value as algo.name for algo in ctrl.compressionAlgorithms">
+                            </select>
+                            <i class="oui-icon oui-icon-chevron-down" aria-hidden="true"></i>
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div class="oui-field pl-5" data-ng-class="{ 'oui-field_error': ctrl.form.$submitted && ctrl.form.stream_archive_duration.$invalid }">
+                <label for="stream_archive_duration" class="oui-label" data-translate="logs_streams_select_archieve_duration"></label>
+                <div class="oui-field__content">
+                    <div class="oui-field-control oui-field-control_3">
+                        <div class="oui-select"
+                            data-ng-class="{ 'oui-select_error': ctrl.form.$submitted && ctrl.form.stream_archive_duration.$invalid }">
+                            <select class="oui-select__input"
+                                name="stream_archive_duration"
+                                id="stream_archive_duration"
+                                data-ng-model="ctrl.stream.data.coldStorageRetention"
+                                data-ng-options="duration.value as duration.name for duration in ctrl.storageDurations">
+                            </select>
+                            <i class="oui-icon oui-icon-chevron-down" aria-hidden="true"></i>
+                        </div>
+                    </div>
+                </div>
+            </div>
+
+            <div class="oui-field pl-5">
+                <label class="oui-label" data-translate="logs_streams_notification_title"></label>
+                <div class="oui-field__content">
+                    <div class="oui-field-control">
+                        <oui-checkbox model="ctrl.stream.data.coldStorageNotifyEnabled" text="{{:: 'logs_streams_notification_detail' | translate}}"></oui-checkbox>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <div class="cui-form-actions">
+            <oui-spinner data-ng-show="ctrl.saving.loading"></oui-spinner>
+            <div data-ng-hide="ctrl.saving.loading">
+                <button type="submit"
+                    class="oui-button oui-button_primary"
+                    data-translate="common_edit"
+                    data-ng-click="ctrl.updateStream()"
+                    data-ng-show="ctrl.isEdit"></button>
+                <button type="submit"
+                    class="oui-button oui-button_primary"
+                    data-translate="common_add"
+                    data-ng-click="ctrl.createStream()"
+                    data-ng-hide="ctrl.isEdit"></button>
+                <a class="oui-button oui-button_link"
```

Comment:
You should use NavigationService.previousState or $window.history.back()

Reply:
Back button must always take user back to streams home page. So I can not use NavigationService.previousState.go() or $window.history.back().

When user visits edit stream page from bookmark url, using $window.history.back() will break

Reply:
Answer in previous comment